build/package/Microsoft.DotNet.Cli.Archive.targets

```diff
@@ -20,19 +21,24 @@
     </ItemGroup>
   </Target>
 
-  <Target Name="GenerateArchives" 
-          DependsOnTargets="Init;Layout;SetupGenerateArchivesInputsOutputs" 
-          Inputs="%(GenerateArchivesInputsOutputs.Inputs)" 
+  <Target Name="GenerateArchives"
+          DependsOnTargets="Init;Layout;SetupGenerateArchivesInputsOutputs"
+          Inputs="%(GenerateArchivesInputsOutputs.Inputs)"
           Outputs="%(GenerateArchivesInputsOutputs.Outputs)">
 
-    <ArchiveDirectory
-        FileName="%(GenerateArchivesInputsOutputs.OutFileName)"
-        OutputDirectory="$(ArchiveOutputDirectory)"
-        InputDirectory="%(GenerateArchivesInputsOutputs.InputDirectory)" >
+    <ZipFileCreateFromDirectory
+        Condition=" '$(OSName)' == 'win' "
+        SourceDirectory="%(GenerateArchivesInputsOutputs.InputDirectory)"
+        DestinationArchive="$(ArchiveOutputDirectory)/%(GenerateArchivesInputsOutputs.OutFileName)" />
+
+    <TarGzFileCreateFromDirectory
+        Condition=" '$(OSName)' != 'win' "
+        SourceDirectory="%(GenerateArchivesInputsOutputs.InputDirectory)"
+        DestinationArchive="$(ArchiveOutputDirectory)/%(GenerateArchivesInputsOutputs.OutFileName)" />
 
-        <Output TaskParameter="OutputArchive"
-                ItemName="Archives" />
-    </ArchiveDirectory>
+    <ItemGroup>
+      <Archives Include="$(ArchiveOutputDirectory)/%(GenerateArchivesInputsOutputs.OutFileName)" />
```

Comment:
This means a PR in build tools. I can look at it once I start porting the targz task there.